user/howto/add-elk-stack-to-an-application.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: user-doc
+title: Add ELK Stack to an Application
+---
+
+This page details how to add [Elasticsearch](https://www.elastic.co/products/elasticsearch),
+[Logstash](https://www.elastic.co/products/logstash) and [Kibana](https://www.elastic.co/products/kibana)
+- the Elastic stack or ELK stack - to an application.
+
+# Elasticsearch Setup
+
+1. Add a new platform for Elasticsearch using the 'Elasticsearch with LB' pack.
+
+1. If required, edit the configuration under the Elasticsearch component.
+
+1. Commit the design changes and deploy the new platform.
+
+1. Once Elasticsearch deployed successfully, you can access the user interface at http://ipaddress:9200
+
+
+# Kibana Setup
+
+1. Add a new platform for Kibana and a dependency to the Elasticsearch platform.
+
+1. Configure the Kibana component pointing to the Elasticsearch component deployed above.
+
+1. Commit the design changes and deploy Kibana.
+
+1. Verify Kibana by accessing the user interface at http://ipaddress:5601/app/kibana
+
+
+# Logstash Setup
+
+The following steps are an example on how to configure Logstash to collect the Tomcat access log.
+
+1. Add a Logstash component under the Tomcat platform.
+
+1. Edit the inputs, filters and outputs options as required.
```

Comment:
`Inputs : file {path => "/opt/tomcat7/logs/access*.log"  sincedb_path => "/opt/logstash/sincedb-access"  }`

`Filters:  grok {     match => {       "message" => "%{IP:client_ip} %{USER:ident} %{USER:auth} \[%{HTTPDATE:apache_timestamp}\] \"%{WORD:method} /%{NOTSPACE:request_page} HTTP/%{NUMBER:http_version}\" %{NUMBER:server_response}"}}`

`Outputs:  if "_grokparsefailure" not in [tags] {     stdout {       codec => rubydebug     }     elasticsearch { host => ["ipaddress"]  protocol => http  port => "port"  }   } `